content/en/apps/guides/messaging/rapidpro.md

```diff
@@ -0,0 +1,114 @@
+
+- set the body to relay the message to CHT in the expected format: 
+```json
+@(json(object(
+  "id", run.uuid,
+  "from", replace(urns.tel,"tel:+", "+"),
+  "content", input.text
+)))
+```
+
+![flow_webhook_body](flow_webhook_body.png)
+
+
+### Create a new trigger
+Create a trigger (`/trigger/`) to start the new flow when a message is not handled elsewhere. 
+
+![trigger_select](trigger_select.png)
+
+For more details about RapidPro configuration, please consult the [RapidPro integration documentation]({{% ref "apps/guides/integrations/rapidpro" %}}). 
+
+## CHT Core configuration
+
+### API keys
+
+The API keys should be treated as securely as a password as anyone with access to them will be able to send messages using your account. Therefore, we store them in the CouchDB configuration.
+
+To add the credentials to the admin config you need to either [PUT the value using curl](https://docs.couchdb.org/en/stable/api/server/configuration.html#put--_node-node-name-_config-section-key) or similar:
+
+```sh
+curl -X PUT https://<user>:<pass>@<domain>/_node/couchdb@127.0.0.1/_config/medic-credentials/rapidpro:incoming -d '"<cht_api_key>"'
+curl -X PUT https://<user>:<pass>@<domain>/_node/couchdb@127.0.0.1/_config/medic-credentials/rapidpro:outgoing -d '"<rapidpro_api_key>"'
+```
+
+{{% alert title="Note" %}}
+`couchdb@127.0.0.1` is the local node name, and may be different for you depending on your setup.
+{{% /alert %}}
+
+You can also add it via Fauxton:
+- Navigate to [the Config screen](http://localhost:5984/_utils/#/_config)
+- Click `Add Option`
+- The `Section` should be `medic-credentials`, the `Name` should be `rapidpro:incoming` or `rapidpro:outgoing`, and the value should be the relevant API key.
+- Click `Create`
+- You should then be able to see your credential in the list of configuration shown
+
+### App settings
+
+Update your app settings as follows.
+
+```json
+{
+  "sms": {
+    "outgoing_service": "rapidpro",
+    "rapidpro": {
+      "url": "<RapidPro instance url>"
+    }
+  }
+}
+```
+
+{{% alert title="Rate limiting" %}}
+The RapidPro API endpoints, which this integration relies on, are rate-limited. The number of messages that can be sent and the number of status updates that can be requested is limited. Check [RapidPro API reference](https://rapidpro.io/api/v2/) for more details.
+
+If an action (sending a message or retrieving a status update) fails, it will be retried again later.
+{{% /alert %}}
+
+{{% alert title="Caveat when switching outgoing message services" %}}
+When switching from another outgoing message service to RapidPro, or switching between RapidPro accounts or hosts, make sure that none of the outgoing messages on your instance, that have been relayed through the previous service, are in a non-final state ("received-by-gateway", "forwarded-by-gateway" or "sent"). CHT Core will request state updates for these messages from RapidPro, fail on every retry, and consume your request quota.  
```

Comment:
Slight rewording for readability...

```suggestion
When switching from another outgoing message service to RapidPro, or switching between RapidPro accounts or hosts, make sure that all of the outgoing messages on your instance, that have been relayed through the previous service, are in a final state ("delivered" or "failed"). CHT Core will request state updates for non-final messages from RapidPro, which will fail on every retry and consume your request quota.  
```

Reply:
This is how I worded it initially, actually. But I changed my mind when I had to list all the other "final" state we have on the CHT: "muted", "denied", "duplicate", "cleared". I should list all, correct? These messages are not used by RapidPro, but could be used by other services.

Reply:
Ahh right. Yeah that's a long list. I'll leave it to you to either ignore this suggestion or list all final states - whichever you think is best.

Reply:
I've listed all final states :)